pkg/yurthub/proxy/pool/pool.go

```diff
@@ -243,14 +243,14 @@ func (pp *PoolCoordinatorProxy) cacheResponse(req *http.Request, resp *http.Resp
 	if pp.localCacheMgr.CanCacheFor(req) {
 		ctx := req.Context()
 		req = req.WithContext(ctx)
-		wrapPrc, _ := hubutil.NewGZipReaderCloser(resp.Header, resp.Body, req, "cache-manager")
+		rc, prc := hubutil.NewDualReadCloser(req, resp.Body, true)
+		wrapPrc, _ := hubutil.NewGZipReaderCloser(resp.Header, prc, req, "cache-manager")
```

Comment:
would you like to explain the reason  of changing the order of gzip and dualreader?

Reply:
It seems no need to change. I'll fix it.

Reply:
Though it has no effect on execution, it seems that it's more readable if we change the order. The 3rd parameter of `NewDualReadCloser` is `isRespBody`. If we use the original code

```go
wrapPrc, _ := hubutil.NewGZipReaderCloser(resp.Header, resp.Body, req, "cache-manager")
rc, prc := hubutil.NewDualReadCloser(req, wrapPrc, false)
```

It does not obviously that if the wrapPrc is response body or not, thus it cause the mistaken 3rd argument as `false` which should be `true` in fact.

While if we use the modified order, we can avoid the mistake. It's obvious that we should set it as `true`.
```go
rc, prc := hubutil.NewDualReadCloser(req, resp.Body, true)
```

Reply:
@Congrool we need to use GZIP reader before dualReader, so all readers in dualReader can get the response data that has been unpacked by GZIP reader.

Reply:
OK, I'll fix it